### PR TITLE
Ephemeral voting key

### PR DESCRIPTION
**Pull request by fboucquez**
Ephemeral Private Voting Key
Replaced PROMPT_MAIN_VOTING with PROMPT_MAIN_TRANSPORT
Improved logging and user feedback when asking keys and generating new voting key files

### CODE REVIEW

File: docs/presetGuides.md [hunk 1]
```diff
@@ -280,6 +280,9 @@ Usage examples:
 
 If your custom preset contains private keys, it's highly recommended to `encrypt` it and provide `--password` when starting or configuring the node with Bootstrap.
 
+Note: Voting keys are ephemeral. They cannot be provided, bootstrap will always generate a new one when resetting the configuration. Bootstrap will only store the public key of the voting key for reference and linking.
+If the voting key file in the target folder gets lost, bootstrap will generate a new voting key file a new random key, and you would need to relink it to your main account.
```
Comment (segfaultxavi) [inline docs/presetGuides.md:284]:
```suggestion
If the voting key file in the target folder is lost, bootstrap will generate a new voting key file from a new random key, and you will need to relink it to your main account (`link` command).
```

File: docs/presetGuides.md [hunk 2]
```diff
@@ -328,7 +331,7 @@ If this is you case, you can tell Bootstrap to not store these keys with the `pr
...
 -   `ENCRYPT`: All private Keys are encrypted and stored in the target's `addresses.yml` file. Bootstrap will have them to be used when required. This is Bootstrap's default behaviour.
 -   `PROMPT_MAIN`: Main Private Keys are not stored in the target's `addresses.yml` file. Bootstrap will prompt for the Main Private Key when generating certificates, or transactions need to be signed in the `link` and `enrolProgram` commands.
--   `PROMPT_MAIN_VOTING`: Main and voting private keys are not stored in the target's `addresses.yml` file. Bootstrap will request the main private key when certificates are generated, or transactions need to be signed by the `link` and `enrolProgram` commands. The voting private key will be requested when generating the voting key file.
+-   `PROMPT_MAIN_TRANSPORT`: Main and transport private keys are not stored in the target's `addresses.yml` file. Bootstrap will request the main private key when certificates are generated, or transactions need to be signed by the `link` and `enrolProgram` commands. The transport private key would be asked when upgrading supernode agents.
```
Comment (segfaultxavi) [inline docs/presetGuides.md:334]:
```suggestion
-   `PROMPT_MAIN_TRANSPORT`: Main and transport private keys are not stored in the target's `addresses.yml` file. Bootstrap will request the main private key when certificates are generated, or transactions need to be signed by the `link` and `enrolProgram` commands. The transport private key will be asked when upgrading supernode agents.
```

File: src/service/VotingService.ts [hunk 3]
```diff
@@ -54,13 +53,11 @@ export class VotingService {
                 logger.info(`Voting File for node ${nodePreset.name} has been previously generated. Reusing...`);
                 return;
             }
+            const votingPrivateKey = nodeAccount?.voting.privateKey;
+            if (!votingPrivateKey) {
+                throw new Error('Voting key should have been previously generated!!!');
```
Comment (segfaultxavi) [inline src/service/VotingService.ts:58]: Can we tell the user what to do in this case?

**Review by segfaultxavi — CHANGES_REQUESTED** [review]

File: src/service/VotingService.ts (hunk 3 above)
Comment (fboucquez) [reply src/service/VotingService.ts]: Updated, atm they need to hard reset the target folder using the original custom preset. It should be an odd case where the voting key file got lost

**Review by Wayonb — APPROVED** [review]

**Pull request merged**